setup.py

```diff
@@ -1,9 +1,21 @@
 #! /usr/bin/env python
 # -*- coding: utf-8 -*-
 
+import os
 
 from setuptools import setup, find_packages
 
+openfisca_core_version = " >=31.0,<33.0"
+local_requires = [
+    "OpenFisca-Core" + openfisca_core_version
+    ]
+server_requires = [
+    "OpenFisca-Core[web-api]" + openfisca_core_version,
+    "OpenFisca-Tracker >=0.4.0,<1.0.0"
+    ]
+server = os.environ.get('SERVER', None)
```

Comment:
Je ne crois pas que ce soit une bonne idée d'utiliser des variables d'environnement dans un setup.py. Si je me souviens bien, cette variable d'environnement va être évaluée lors de la _publication_ de la librairie sur Pypi, et pas à l'_installation_. 
Les dépendances de France ne dépendront pas donc de l' environnement dans lequel il est installé, mais de celui depuis lequel il a été publié, ce qui n'est sans doute pas le comportement désiré.

Reply:
@fpagnoux Oui j'avais conscience du contexte d'exécution, et cette implémentation peut surprendre mais elle me semble à la fois sans risque et nécessaire pour atteindre l'objectif "déployer sur Scalingo".

"L'environnement dans lequel OpenFisca-France est publié" c'est nous qui le déterminons, et actuellement c'est l'environnement CircleCI tel que décrit dans notre config; dans cet environnement la variable `SERVER` n'est pas renseignée, donc de façon tout à fait fiable OpenFisca-France est publié *sans* les dépendances `web-api` de Core.

"L'environnement dans lequel OpenFisca-France est installé" c'est plus variable, il y trois cas de figure:
- l'installation du paquet depuis PyPI avec par exemple `pip install openfisca-france`; dans ce cas de figure on va installer les dépendances déterminées à la publication donc *sans* `web-api`, comme actuellement;
- l'installation en local pour le développement: `git clone openfisca-france && install -e.`, dans ce cas de figure, par défaut la variable `SERVER` n'est pas renseignée (*), et on installe donc OpenFisca-France *sans* `web-api`, comme actuellement;
- l'installation en production, pour exploiter l'API; elle s'appuie également sur un checkout du code avec `git clone` que ce pour notre VPS ou pour Scalingo. Ce que cette PR apporte c'est une façon "[12-factor](https://12factor.net/fr/)" de stipuler que nous avons besoin de Core *avec* `web-api`; sur Scalingo nous n'avons *que* ce moyen de moduler le comportement à l'installation depuis une copie du dépôt.

(*) par contre ce choix de nom de variable est peut-être trop vulnérable à des collisions, je peux choisir un nom plus spécifique comme OPENFISCA_SERVER ou SERVER_DEPS (aurais-tu une préférence ?)

Reply:
`OPENFISCA_INSTALL_WEB_API`, to be super explicit?

Reply:
I still don't know how to fill about putting that here though... A `setup.py` describes the dependencies of a library, not an application, so it feels a little out of place to invoke 12factors here, and to couple OpenFisca-France requirement as a library to a server environment. 

At the same time, I know pip is bad at managing dependencies, and we haven't so far found an elegant solution to the problem I think this PR is going to solve.

Reply:
Moi j'ai l'impression que ce que nous deployons actuellement, c'est plutôt une _combinaison de librairies_ avec du code applicatif pour les faire tourner.

Plus concrètement, le module `web-api` d'OpenFisca-Core avec le module `OpenFisca-France`.

Vu que le module `web-api` n'appartient pas à OpenFisca-France, il m'a frappé justement que dans cette pull request on parle de `12factors`, qui s'applique aux web apps.

Ceci n'est pas une web app mais une librairie. On peut effectivement trouver un moyen de déployer une web app qui réutilise ce module et le commiter même ce même dépôt, mais amha ces éléments n'appartiennent pas dans le setup de cette librairie.